src/solver/simulation/sim_calcul_economique.cpp

```diff
@@ -147,9 +147,15 @@ void SIM_InitialisationProblemeHebdo(Data::Study& study,
         break;
     }
 
+    Antares::Data::Area::ScratchMap scratchmap;
+    study.areas.each([&scratchmap, &numspace](Antares::Data::Area& a) {
+            scratchmap.try_emplace(&a, a.scratchpad[numspace]); });
```

Comment:
You could define a small wrapper around the map that would allow you to just write
```suggestion
    Antares::Data::Area::ScratchMap scratchmap(study.areas);
```
ie you transfer the responsability of the creation to the constructor of `ScratchMap`